sw/device/tests/sysrst_ctrl_ulp_z3_wakeup_test.c

```diff
@@ -15,13 +15,19 @@
 #include "sw/device/lib/testing/rstmgr_testutils.h"
 #include "sw/device/lib/testing/test_framework/check.h"
 #include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/lib/testing/test_framework/ottf_utils.h"
 
 #include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
 
-OTTF_DEFINE_TEST_CONFIG();
+/* We need control flow for the ujson messages exchanged
+ * with the host in OTTF_WAIT_FOR on real devices. */
+OTTF_DEFINE_TEST_CONFIG(.enable_uart_flow_control = true);
 
-// This is updated by the sv component of the test
-static volatile const uint8_t kTestPhase[1];
+// This is updated by the sv component of the test.
+// On DV, we must use variables in flash but on a real device,
+// we must use variables in RAM.
+static volatile const uint8_t kTestPhaseDV[1];
+static volatile uint8_t kTestPhaseReal = 0;
```

Comment:
It's not const:
```suggestion
static volatile uint8_t test_phase_real = 0;
```

Reply:
This variable is a bit weird: like `kTestPhaseDV`, it is backdoor-written to by the host using the OTTF and so must not be written by the device code. But it is not declared `const` so that the compiler will put it in RAM and not flash. I guess alternatively I could do
```c
// This variable must go to RAM so that the host can change it's value using the OTTF.
OT_SECTION(".data")
static volatile const uint8_t kTestPhaseReal = 0;
```
if you think that's clearer?

Reply:
I think I'd probably go for that change, since it's a bit more explicit (and doesn't require someone to have the "const vs non-const" / linker script rules in their head!)